mono/eglib/goutput.c

```diff
@@ -274,3 +276,23 @@ g_set_printerr_handler (GPrintFunc func)
 	return old;
 }
 
+static void
+unity_vprintf_GPrintFunc_adapter (const gchar *string)
+{
+	our_vprintf("%s", string);
```

Comment:
I think you need an intermediate wrapper
```
void wrap_our_vprintf(const gchar *format, ...)
{
	va_list args;
	va_start (args, format);
    our_vprintf(format, args);
	va_end (args);
}
```

And then have `unity_vprintf_GPrintFunc_adapter` and `unity_vprintf_GLogFunc_adapter` call `wrap_our_vprintf` to properly setup the `va_list`